pages/api/auth/[...nextauth].ts

```diff
@@ -15,10 +15,36 @@ export const authOptions: AuthOptions = {
       }
       return false;
     },
-    async jwt({ token, user }) {
-      if (user && user?.accessToken) {
-        return { ...token, user };
+    async jwt({ token }) {
+      if (token?.exp && Date.now() < token.exp * 1000) {
+        try {
+          const response = await fetch(
+            `${API_BASE_URL}/dj-rest-auth/token/refresh/`,
+            {
+              headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+              body: new URLSearchParams({
+                refresh: token.user.refreshToken,
+              }),
+              method: 'POST',
+            }
+          );
+
+          const tokens = await response.json();
+
+          if (!response.ok) throw tokens;
+
+          return {
+            ...token,
+            user: { ...token.user, accessToken: tokens.access },
+            exp: new Date(tokens.access_token_expiration).getTime(),
+          };
+        } catch (error) {
+          // Return previous token on error
+          console.log(error);
```

Comment:
Can you return de error or some error message instead of login it? Something more useful for the user. WDYT?

Reply:
This is ran on the server, so the user won't actually see anything, even the `console.log`s are only shown on my terminal and not on the browser's console. 

I don't really understand NextAuth all that much, but I believe whatever I return here is just saved as the token by NextAuth.